src/core/src/bound_evaluate.cpp

```diff
@@ -75,85 +75,219 @@ bool is_type_allocable(const element::Type& type) {
+        const auto ub = tensor_desc.get_upper_value();
+        return static_cast<bool>(lb) != static_cast<bool>(ub) || !lb;
+    }
 
-                    if (tensor.get_upper_value() && shape_size(tensor.get_upper_value().get_shape()) > 10)
-                        should_invalidate |= true;
+protected:
+    void init(const Node& node) {
+        lowers.resize(0);
+        uppers.resize(0);
 
-                    if (should_invalidate && input.get_target_inputs().size() == 1)
-                        tensor.invalidate_values();
-                }
-                propagate_rt_info(node, output);
+        for (const auto& output : node.outputs()) {
+            if (is_type_allocable(output.get_element_type())) {
+                lowers.emplace_back(output);
+                uppers.emplace_back(output);
             } else {
-                break;
+                lowers.emplace_back();
+                uppers.emplace_back();
+            }
+        }
+    }
+
+    void set_bound_at_port(descriptor::Tensor& tensor_desc, const size_t port, const bool same_inputs) const {
+        tensor_desc.set_lower_value(lowers[port]);
+        if (same_inputs || are_equal(lowers[port], uppers[port])) {
+            tensor_desc.set_upper_value(lowers[port]);
+        } else {
+            tensor_desc.set_upper_value(uppers[port]);
+        }
+    }
+
+    bool evaluate(const Node& node) {
+        return node.evaluate_lower(lowers) && node.evaluate_upper(uppers);
+    }
+
+    ov::TensorVector lowers, uppers;
+};
+
+/** @brief BoundEvaluator evaluate and set bound depends specified BoundType. */
+template <class BoundType>
+struct BoundEvaluator : BoundType {
+    void init(const Node* n) {
+        node = n;
+        symbols.resize(0);
+        BoundType::init(*node);
+    }
+
+    void set_bounds_and_symbols() const {
+        const auto same_inputs = node_has_same_inputs(*node);
+        const auto symbols_evaluated = !symbols.empty();
+
+        for (size_t port = 0; port < node->get_output_size(); ++port) {
+            auto& output_tensor_desc = node->get_output_tensor(port);
+
+            BoundType::set_bound_at_port(output_tensor_desc, port, same_inputs);
+
+            if (symbols_evaluated && output_tensor_desc.get_value_symbol().empty()) {
+                output_tensor_desc.set_value_symbol(symbols[port]);
+                ov::populate_tensor_with_missing_symbols(output_tensor_desc);
+            }
+        }
+    }
+
+    bool evaluate() {
+        if (BoundType::evaluate(*node)) {
+            symbols.resize(node->get_output_size());
+            if (!node->evaluate_symbol(symbols)) {
+                symbols.resize(0);
             }
+            return true;
+        } else {
+            return false;
         }
     }
 
-    if (is_upper)
-        return output.get_tensor().get_upper_value();
-    else
-        return output.get_tensor().get_lower_value();
+    const Node* node;
+    TensorSymbolVector symbols;
```

Comment:
`const Node* node{};
    TensorSymbolVector symbols{};`

Reply:
I shouldn't be required will be default initialized when default constructor will be used.